src/csharp/examples/server/API/DoCoolStuff/DoCoolStuffController.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.AspNetCore.Mvc;
+using System;
+using SessionlessExample.Server.Models;
+using System.Text.Json;
+using SessionlessNET.Impl;
+using SessionlessNET.Models;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SessionlessExample.Server.Controllers
+
+{
+    [ApiController]
+    [Route("/cool-stuff")]
+    public class DoCoolStuffController: ControllerBase
+    {
+        Vault Vault;
+        Sessionless Sessionless;
+        const string privPath = "./priv.key";
+        const string pubPath = "./pub.key";
+        static void VaultSaver(KeyPairHex pair) {
+            
+        }
+        static KeyPairHex? VaultGetter() {
+            
+            return new("", "");
+        }
+
+        [HttpPost]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        public IActionResult PostJson(Models.DoCoolStuffModel input) {
+            Vault = new Vault(VaultGetter, VaultSaver);
+            Sessionless = new(Vault);
+
+            UserModel user = DemoPersistenceController.getUser(input.uuid);
+
+            if(user == null) {
```

Comment:
Note in C#:
- use `x is null` if you want to really know if something is null or not
- use `x == null` if you want to use the `==` operator of the type (which uses `Equals` in most cases) which allow for custom returns by the class

### The safe way:
use `x is null` in most cases since it ensures whether `x` is null and lets you completely avoids NRE